okhttp/src/main/java/okhttp3/internal/http2/Http2Connection.java

```diff
@@ -620,7 +620,7 @@ public Http2Connection build() {
           // Create a stream.
           final Http2Stream newStream = new Http2Stream(streamId, Http2Connection.this,
               false, inFinished, headerBlock);
-          lastGoodStreamId = streamId;
+          if (streamId > lastGoodStreamId) lastGoodStreamId = streamId;
```

Comment:
I don’t think this works. The lastGoodStreamId refers to the _local_ last good stream, and this is the _remote_ last good stream